### ISSUE TITLE
False positives with the `gci` linter

### ISSUE THREAD

**Issue by magnetikonline** [golangci/golangci-lint#3275]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

Seeing the following false positives when using the `gci` linter:

```
$ golangci-lint run
path/to/main/main.go:5: File is not `gci`-ed with --skip-generated -s standard,default (gci)
Usage
```

This was also happening with release `v1.49` - version `v1.48` is the last known working release with the same `.golangci.yml` and source code using `gci`.

If I update the offending line of `Usage` to `Usage:` or `Usage blah` (more than a single word) - it passes. Note that from the code example this is a _comment_ at the top of a package, before the `package main` line.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.50.0 built from 704109c6 on 2022-10-04T10:25:07
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
run:
  timeout: 2m

linters:
  enable:
    - errname
    - exhaustive
    - exportloopref
    - gci
    - goconst
    - godot
    - goimports
    - unconvert
    - whitespace

linters-settings:
  errcheck:
    exclude-functions:
      - (net/http.ResponseWriter).Write
  exhaustive:
    default-signifies-exhaustive: true
  goconst:
    min-len: 2
    min-occurrences: 2

issues:
  exclude-rules:
    - path: _test\.go$
      linters:
        - goconst
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.18.7 darwin/arm64
GO111MODULE=""
GOARCH="arm64"
GOBIN=""
GOCACHE="/Users/magnetik/Library/Caches/go-build"
GOENV="/Users/magnetik/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/magnetik/go/pkg/mod"
GONOPROXY="github.com/myorg"
GONOSUMDB="github.com/myorg"
GOOS="darwin"
GOPATH="/Users/magnetik/go"
GOPRIVATE="github.com/myorg"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.18.7"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/dev/null"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/_q/w0qy822d3pvcdvdq21l9drq40000gn/T/go-build3104308243=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/magnetik/project/my-project /Users/magnetik/project /Users/magnetik /Users /]
INFO [config_reader] Used config file .golangci.yml
INFO [lintersdb] Active 16 linters: [errcheck errname exhaustive exportloopref gci goconst godot goimports gosimple govet ineffassign staticcheck typecheck unconvert unused whitespace]
INFO [loader] Go packages loading at mode 575 (deps|exports_file|imports|types_sizes|compiled_files|files|name) took 445.334208ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 8.508667ms
INFO [linters_context/goanalysis] analyzers took 34.290257681s with top 10 stages: buildir: 17.454555741s, unconvert: 1.808820214s, exhaustive: 1.452458546s, ctrlflow: 1.224663192s, fact_deprecated: 1.181795834s, gci: 1.086795918s, printf: 995.74401ms, inspect: 864.402401ms, nilness: 600.730834ms, goimports: 488.641ms
INFO [runner] Issues before processing: 121, after processing: 1
INFO [runner] Processors filtering stat (out/in): uniq_by_line: 1/2, max_from_linter: 1/1, source_code: 1/1, severity-rules: 1/1, path_prefixer: 1/1, cgo: 121/121, filename_unadjuster: 121/121, identifier_marker: 50/50, exclude-rules: 22/50, max_per_file_from_linter: 1/1, path_shortener: 1/1, path_prettifier: 121/121, skip_files: 121/121, autogenerated_exclude: 50/121, exclude: 50/50, diff: 1/1, max_same_issues: 1/1, skip_dirs: 121/121, nolint: 2/22, sort_results: 1/1
INFO [runner] processing took 6.267785ms with stages: nolint: 3.534917ms, exclude-rules: 1.060125ms, identifier_marker: 580µs, path_prettifier: 528.042µs, autogenerated_exclude: 393.625µs, skip_dirs: 133.417µs, source_code: 18.625µs, cgo: 8.708µs, filename_unadjuster: 4.375µs, uniq_by_line: 1.457µs, max_same_issues: 999ns, max_from_linter: 875ns, max_per_file_from_linter: 708ns, path_shortener: 541ns, skip_files: 416ns, severity-rules: 249ns, exclude: 208ns, diff: 208ns, sort_results: 208ns, path_prefixer: 82ns
INFO [runner] linters took 4.395439709s with stages: goanalysis_metalinter: 4.389117792s
path/to/main/main.go:5: File is not `gci`-ed with --skip-generated -s standard,default (gci)
Usage
INFO File cache stats: 2 entries of total size 197.9KiB
INFO Memory: 50 samples, avg is 805.9MB, max is 1548.7MB
INFO Execution took 4.857086792s
```

</details>

### Code example or link to a public repository

<details>

```go
/*
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vitae nisl arcu.
Mauris eu quam tempor, cursus ipsum a, placerat turpis. Maecenas eleifend arcu ut sem mattis.

Usage

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vitae nisl arcu.
Mauris eu quam tempor, cursus ipsum a, placerat turpis. Maecenas eleifend arcu ut sem mattis,
*/
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/aws/aws-sdk-go/aws/session"
)
```

</details>

**Comment by ldez** [comment]
Hello,

> [x] Yes, I've tried with the standalone linter if available

As you have checked this, can you give me the output of the [gci](https://github.com/daixiang0/gci) command line?

**Comment by ldez** [comment]
I take your configuration and your sample, and I don't reproduce the problem.

<details>
```go
/*
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vitae nisl arcu.
Mauris eu quam tempor, cursus ipsum a, placerat turpis. Maecenas eleifend arcu ut sem mattis.

# Usage

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vitae nisl arcu.
Mauris eu quam tempor, cursus ipsum a, placerat turpis. Maecenas eleifend arcu ut sem mattis,
*/
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/aws/aws-sdk-go/aws/session"
)

func Foo(ctx context.Context) {
	fmt.Println("")
	os.Getenv("JI")
	_ = session.Session{}
}

```


```console
$ golangci-lint version  
golangci-lint has version 1.50.0 built from 704109c6 on 2022-10-04T10:25:07Z
$ golangci-lint run 
$ 

</details>

**Comment by magnetikonline** [comment]
> As you have checked this, can you give me the output of the [gci](https://github.com/daixiang0/gci) command line?

Ah sorry @ldez - just ran it over the package in question - no error:

```
$ gci diff .
```

if I deliberately munge the package order I do get an error, so the linter is working:

```
$ gci diff .
--- my-package/main.go
+++ my-package/main.go
@@ -12,6 +12,7 @@
 import (
 	"context"
 	"fmt"
+	"os"

 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/foo/bah/v2"
@@ -27,8 +28,6 @@
 	"github.com/foo/bah/log"
 	"github.com/foo/bah/zzz"
-
-	"os"
 )

 const myConst = "value"
```

**Comment by magnetikonline** [comment]
> I take your configuration and your sample, and I don't reproduce the problem.

Did you have `# Usage` or `Usage` as that comment line?

**Comment by ldez** [comment]
> Did you have # Usage or Usage as that comment line?

yes go fmt tricked me.

**Comment by magnetikonline** [comment]
Okay - not a problem, will have to stick with `1.47.3` for now. 👍

**Comment by ldez** [comment]
I think your problem is related to a side effect of go fmt formatting requirements but I don't know why.

**Comment by magnetikonline** [comment]
@ldez that's a good point - maybe a bump to Golang `1.19` might help here - something I'm looking at doing in the near future 👍.

**Comment by ldez** [comment]
```
$ golangci-lint run --disable-all -Egofmt
foo.go:5: File is not `gofmt`-ed with `-s` (gofmt)
Usage
```

**Comment by magnetikonline** [comment]
Ahhh - nice one @ldez 👍 - that does smell very similar 👍

**Comment by ldez** [comment]
I think know: it's because gci uses the format API (the basic API of `go fmt`)

**Comment by magnetikonline** [comment]
Great - I'll have to circle back on this after a bump to Golang `1.19` 👍

**Comment by ldez** [comment]
As we know the root cause, and because I don't really think that spending time on fixing a problem of an "old" Go version has a good ROI, can we close the issue?

**Comment by magnetikonline** [comment]
Happy with that.... close it. 👍

**Comment by magnetikonline** [comment]
> I think know: it's because gci uses the format API (the basic API of `go fmt`)

@ldez just to circle back - starting porting some projects to Golang `1.19` and using the latest `golangci-lint` release - and everything is good now. It's certainly related to the `go fmt` changes that `1.19` now applies on Godoc comment blocks that was corrupting the linting results due to shifted line numbers/etc.
